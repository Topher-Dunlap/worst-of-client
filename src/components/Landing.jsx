import React, {useContext} from 'react';
import ThemeContext from "./ThemeContext";

export default function Projects() {

    const context = useContext(ThemeContext);
    const theme = context.theme;

    return (
        <section style={theme}>
            <div>
                <h1 style={bottomMargin}>Landing</h1>
            </div>
        </section>
    )
}

const bottomMargin = {
    marginBottom: "2rem",
}
