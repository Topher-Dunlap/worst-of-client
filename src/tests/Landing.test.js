import React from 'react';
import ReactDOM from 'react-dom';
import Landing from '../components/Landing';
import ThemeContext from "../components/ThemeContext";
import renderer from 'react-test-renderer';

describe(`Landing component`, () => {

    const backgroundColors = [
        {backgroundColor: "#AAAAAA"},
        {backgroundColor: "#BBBBBB"},
        {backgroundColor: "#EEEEEE"},
    ]

    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <ThemeContext.Provider value={{backgroundColors}}>
                <Landing/>,
            </ThemeContext.Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(
                <ThemeContext.Provider value={{backgroundColors}}>
                    <Landing/>,
                </ThemeContext.Provider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })

})