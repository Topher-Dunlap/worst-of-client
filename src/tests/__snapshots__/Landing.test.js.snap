// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders the UI as expected 1`] = `
Array [
  <section>
    <div>
      <span>
        <h1
          style={
            Object {
              "marginTop": "8rem",
            }
          }
        >
          Worst Of
        </h1>
        <h2>
          Find adventure in any City.
        </h2>
      </span>
      <section>
        <header
          style={
            Object {
              "backgroundColor": "#BBBBBB",
            }
          }
        >
          <h3>
            Experience the lowest rated places a city has to offer.
          </h3>
        </header>
        <p>
          [
          <em>
            placeholder for screenshot of a dive bar
          </em>
          ]
        </p>
        <p>
          Worst Of helps you discover the lowest rated, restaurants, bars and coffee shops in your area.
                        Have you and your friends found yourselves with nothing to do on a friday night? Now's your
                        chance at rolling the fun dice on finding a new local favorite that you may love to hate!
        </p>
      </section>
      <section>
        <header
          style={
            Object {
              "backgroundColor": "#EEEEEE",
            }
          }
        >
          <h3>
            Find something to brag about.
          </h3>
        </header>
        <p>
          [
          <em>
            placeholder for screenshot of terrible restaurant
          </em>
          ]
        </p>
        <p>
          Have your ever struggled to come up with interesting conversation with strangers and acquaintances
                        you meet? How about something to talk about with your co-workers on monday? Worst Of is here to
                        provide you with an unforgettable, mediocre to bad experience you'll love telling everyone
                        about.
        </p>
      </section>
      <section>
        <header>
          <h3>
            Start discovering your low standards today
          </h3>
        </header>
      </section>
      <form
        onSubmit={[Function]}
      >
        <header
          style={
            Object {
              "marginBottom": "3rem",
              "marginTop": "6rem",
            }
          }
        >
          <h1>
            Register
          </h1>
        </header>
        <div>
          <label
            htmlFor="first_name"
          >
            First name
          </label>
          <input
            id="first_name"
            name="first_name"
          />
        </div>
        <div>
          <label
            htmlFor="last_name"
          >
            Last name
          </label>
          <input
            id="last_name"
            name="last_name"
          />
        </div>
        <div>
          <label
            htmlFor="email"
          >
            Email
          </label>
          <input
            id="email"
            name="email"
          />
        </div>
        <div>
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            id="password"
            name="password"
          />
        </div>
        <button
          type="submit"
        >
          Sign Up
        </button>
      </form>
      <div
        style={
          Object {
            "borderColor": "#000000",
            "borderWidth": "thin",
            "fontSize": "medium",
            "margin": "6rem 0",
          }
        }
      >
        <h3>
          Password Requirements
        </h3>
        <ul
          style={
            Object {
              "listStylePosition": "inside",
              "textAlign": "center",
            }
          }
        >
          <li>
            Password must contain 1 upper case, lower case, number and special character
          </li>
          <li>
            Password must be longer than 8 characters
          </li>
          <li>
            Password must be less than 72 characters
          </li>
        </ul>
      </div>
    </div>
  </section>,
  ",",
]
`;
